### DISCUSSION TITLE
Modifiability - 1

### DISCUSSION THREAD

**Discussion by changjurhee** [changjurhee/AONE#16] [Polls]
## Quality Attributes

Type | Description | QA Scenario (Stimulus / Source of stiulus / Artifact / Environment / Response / Response measure) | Tatics
 -- | -- | -- | --
Modifiability | Provide the ability for the developer/maintainer to easily add or replace codecs | Changing the codec for video transmission by customer request should be possible within 5 hours.<br>- Video codec change request<br>- Customer<br>- Video codec related software components<br>- During video codec change<br>- System supports the changed video codec<br>- It takes less than 5 hours to change the video codec |

**Comment by ju297** [comment]
Layer Pattern
 - 장점사용하는 layer와 사용 당하는 layer가 확실하게 구분되어 하위 레이어만 교체하는데 용이
 - Layer bridging을 한다면 변경 용이성을 달성할 수 없음

Pipes and Filters Pattern
 - 데이터 스트림을 처리하기에 적합하며 교체 또한 용이
 - 데이터 병렬처리에 대해서는 다소 부적합 할 수 있음

**Comment by changjurhee** [comment]
- Plug-in (Microkernel) Pattern 
  - Benefits : 
    - Plug-ins provide a controlled mechanism to extend a core product and make it useful in a variety of contexts.

**Comment by codekistone** [comment]
split module - The codec to be used in the system must be made into a separate module so that the codec can be changed only by replacing the module.
